### PR TITLE
ROX-23689: Compliance coverage routing and profile selector

### PR DESCRIPTION
**Pull request by bradr5**
## Description

Routing setup for compliance v2 coverage with profile selector tabs.

This establishes the routing configuration for the coverage feature, including the profile selector tabs that are currently based off of a results endpoint. However, the source of these tabs will later be based off of scan configurations to give us a more up to date list of profiles. Otherwise we would need to fetch this information often due to how results trickle in.

Details:

- The `ComplianceProfilesProvider` wraps the `CoveragePage` to supply profile data via react context
- Profile data is only fetched once when visiting the coverage page but is accessible throughout
- The loading and error handling for profile data should only have to be handled by `CoveragePage`
- A redirect handler is used to take the user to the first profile in the list if no profile is given
- For the most part, routing is closely aligned with endpoints required for each page

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Manual testing done for now. Automated tests should be created when the `getComplianceProfilesStats` endpoint is swapped out for desired endpoint that is currently in progress.

* Placeholder buttons used to represent links to/from different sections of the coverage feature
* Clicking "Coverage" in the side navigation should take the user to the first profile fetched
* Loading, no results, and errors are handled gracefully
* Profile data is fetched only once when navigating to `/compliance-enhanced/coverage/` and not refetched through subsequent redirects within the coverage section

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    const { profileScanStats } = useContext(ComplianceProfilesContext);
+
+    const handleToggleChange = (selectedProfile) => {
+        history.push(`${complianceEnhancedCoveragePath}/profiles/${selectedProfile}/${tableView}`);
+    };
+    return (
+        <ToggleGroup aria-label="Toggle for selected profile view">
+            {profileScanStats &&
+                profileScanStats.scanStats.map((profile) => (
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx:21]: Is optional chaining `profileScanStats?.scanStats.map((profile) => (…))` an alternative?

What are pro and con:
* Add `TODO` comments wherever this temporary data structure is used.
* Find in files, when the time comes. Hopefully sooner than later.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesPageHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { PageSection, Text, Title } from '@patternfly/react-core';
+
+import PageTitle from 'Components/PageTitle';
+
+function CoveragesPageHeader() {
+    return (
+        <>
+            <PageTitle title="Compliance coverage" />
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesPageHeader.tsx:9]: When you need a study break, can you add to your doc a proposal for page titles for browser history?
For example,
* Compliance coverage - Profile
* Compliance coverage - Checks
* Compliance coverage - Check
* Compliance coverage - Clusters
* Compliance coverage - Cluster

Access Control is an example similar to the above.

A question which you can either defer or explore is whether Clusters and Cluster are ambiguous if Benchmark is added in the future.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileChecksPage.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                    variant="primary"
+                >
+                    Go to all clusters (ProfileClustersPage)
+                </Button>
+                <Button
+                    onClick={() => {
+                        history.push(
+                            `${complianceEnhancedCoveragePath}/profiles/${profileName}/checks/test`
+                        );
+                    }}
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileChecksPage.tsx:37]: Although `ToggleGroupItem` supports only an imperative `history.push(…)` call, let’s prefer React Router `Link` element whenever possible when we replace these demo stubs with tables, and so on.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesPageHeader.tsx (hunk 2 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesPageHeader.tsx:9]: I guess "Page titles" would also be a good addition to the [Frontend Design Doc checklist](https://docs.google.com/document/d/1g2VnZ_gJLeKAQSWyNiHF_ovf5JwW76B7vbVfaERyoGU/edit)

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React, { useContext } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { ToggleGroup, ToggleGroupItem } from '@patternfly/react-core';
+
+import { complianceEnhancedCoveragePath } from 'routePaths';
+
+import { ComplianceProfilesContext } from './ComplianceProfilesProvider';
+
+function CoveragesToggleGroup({ tableView = 'checks' }: { tableView: string }) {
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx:9]: Do we want to limit the types of strings we want to allow for the `tableView` variable? 

``` tableView: 'checks' | 'controls'```

**Review by sachaudh — APPROVED** [review]
Sweet, looks good to me. This helps us start dividing and conquering ⚔️

**Review by pedrottimark — APPROVED** [review]

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileChecksPage.tsx (hunk 3 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileChecksPage.tsx:37]: Agreed, unfortunately this same component will be used for switching between checks and clusters so we're going to have to do the same there.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx (hunk 4 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx]: It'll be checks and clusters for now but good callout, updated.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx (hunk 1 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx]: Updated

Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesToggleGroup.tsx]: Assuming we'll be changing the context value type, typescript should complain everywhere this is being used.

**Pull request merged**